libraries/consul_service.rb

```diff
@@ -85,10 +93,13 @@ def service_options(service)
         service.directory(new_resource.data_dir)
         service.user(new_resource.user)
         service.environment(new_resource.environment)
-        service.restart_on_update(false)
+        service.restart_on_update(new_resource.restart_on_update)
         service.options(:systemd, template: 'consul:systemd.service.erb')
         service.options(:sysvinit, template: 'consul:sysvinit.service.erb')
         service.options(:upstart, template: 'consul:upstart.service.erb', executable: new_resource.program)
+        new_resource.options.each do |key, value|
```

Comment:
This is necessary to be able to override the template for systemd unit file (for instance to limit the number of possible open files)